### PR TITLE
[Seeds] Support for production data

### LINKED ISSUE #36: [Spaces] Seeder for prod
The test-data is already quite realistic. I think we can steal:
- The list of spaces...
- Logos
- Location

The following is missing:
- Real descriptions (translated to Thai)
- Real banner images (16:9?)

### PR DESCRIPTION
**Pull request by antonyharfield**
## ✅ DoD

- [x] All work is complete
- [x] Migrations: NA
- [x] Documentation: UPDATED
- [x] Release notes: NA
- [x] Issues linked: resolves #36 

## 📝 Summary

- Move `test-data` to `seeds/emu`
- Refactor to be able to support production and dev

### CODE REVIEW

File: admin/src/seeds/index.ts [hunk 1]
```diff
@@ -1,19 +1,23 @@
...
-  if (MODE !== 'emu') return console.warn('This script is exclusively for local testing')
+  if (!process.argv.includes('--yes')) {
+    await productionWarning(__filename)
+  }
   await migrate()
 
+  const banners = await import(`./${MODE}/banners`).then(_ => _.banners).catch(_ => []) as Banner[]
+  const events = await import(`./${MODE}/events`).then(_ => _.events).catch(_ => []) as Event[]
+  const spaces = await import(`./${MODE}/spaces`).then(_ => _.spaces).catch(_ => []) as Space[]
+
```
Comment (antonyharfield) [inline admin/src/seeds/index.ts]: I'm not sure about `_ => _.banners`... is poor practice in JS/TS?

File: admin/package.json [hunk 2]
```diff
@@ -15,7 +15,7 @@
     "backup": "yarn serve \"src/backup/index.ts\"",
     "config:deploy": "yarn serve \"src/config/functions-deploy.ts\"",
     "migrate": "yarn serve \"src/migrations\"",
-    "seed": "yarn serve \"src/test-data\""
+    "seed": "yarn serve \"src/seeds\""
```
Comment (charles-allen) [inline admin/package.json:18]: For production, is it just this?
```
yarn serve "src/seeds" --mode=production
```

File: admin/src/seeds/production/.gitignore [hunk 3]
```diff
@@ -0,0 +1,2 @@
+*
+!.gitignore
```
Comment (charles-allen) [inline admin/src/seeds/production/.gitignore]: No new line at the end? :shocked-face:

File: admin/package.json (hunk 2 above)
Comment (charles-allen) [reply admin/package.json:18]: You literally answered this in the docs!!

File: admin/src/seeds/index.ts (hunk 1 above)
Comment (charles-allen) [reply admin/src/seeds/index.ts]: I think it's poor practice to name something `_` and then use it!! Underscore is typically reserved to mean "I don't use this variable".

I see you problem; the data files export named consts, so you have to reference them. Another option would be to refactor to use a default export:
```
const banners: Banner[] = [...]
export default banners
```
Then you could delete the `then` entirely.

File: admin/src/seeds/index.ts [hunk 4]
```diff
@@ -1,19 +1,23 @@
...
 
-import { banners } from './data/banners'
-import { events } from './data/events'
-import { spaces } from './data/spaces'
-
 const db = context.db
 
 const main = async (): Promise<void> => {
-  if (MODE !== 'emu') return console.warn('This script is exclusively for local testing')
+  if (!process.argv.includes('--yes')) {
```
Comment (charles-allen) [inline admin/src/seeds/index.ts]: I don't like CLI overrides for this warning. You shouldn't be calling this script from CI/CD, so this only creates the risk of pressing up & accidentally re-running on production.

**I prefer just letting the warning do its job: warning you.**

**Review by charles-allen — CHANGES_REQUESTED** [review]
I think we should fix 3 things:
- Merge PR #56 before this, and refresh/update this branch
- Disallow overriding the production warning
- Optional: use `export default` & remove `then`

File: admin/src/seeds/index.ts (hunk 4 above)
Comment (antonyharfield) [reply admin/src/seeds/index.ts]: Done

File: admin/src/seeds/production/.gitignore (hunk 3 above)
Comment (antonyharfield) [reply admin/src/seeds/production/.gitignore]: :shocked-face:

File: admin/src/seeds/index.ts (hunk 1 above)
Comment (antonyharfield) [reply admin/src/seeds/index.ts]: Done

File: admin/src/seeds/index.ts [hunk 5]
```diff
@@ -1,19 +1,27 @@
...
+
 const main = async (): Promise<void> => {
-  if (MODE !== 'emu') return console.warn('This script is exclusively for local testing')
+  await productionWarning(__filename)
+
   await migrate()
 
+  const banners: Banner[] = await load('banners')
+  const events: Event[] = await load('events')
+  const spaces: Space[] = await load('spaces')
```
Comment (charles-allen) [inline admin/src/seeds/index.ts:23]: I _**think**_ you can parallelize this if you like ;)
```
const [banners, events, spaces] = await Promise.all([
  load('banners'),
  load('events'),
  load('spaces')
])
```

**Review by charles-allen — APPROVED** [review]
LET'S GO!

anna

**Pull request merged**